src/robot/libdocpkg/model.py

```diff
@@ -190,3 +228,155 @@ def _arg_to_dict(self, arg):
             'required': arg.required,
             'repr': unicode(arg)
         }
+
+
+class DataTypeCatalog(object):
+
+    def __init__(self):
+        self._enum_catalog = set()
+        self._typed_dict_catalog = set()
+
+    def __iter__(self):
+        for type_dooc in sorted(self._typed_dict_catalog | self._enum_catalog):
+            yield type_dooc
+
+    def __bool__(self):
+        return bool(self._enum_catalog or self._typed_dict_catalog)
+
+    __nonzero__ = __bool__
+
+    @property
+    def enum_catalog(self):
+        return sorted(self._enum_catalog)
+
+    @property
+    def typed_dict_catalog(self):
+        return sorted(self._typed_dict_catalog)
+
+    def add(self, typ):
```

Comment:
Could this be `add(self, *types)` and separate `update` then removed? Would also avoid the problem with `type` as an argument name.

Reply:
The idea was to be kind of compatible with the interface of a set().